tests/open_alchemy/helpers/test_peek/test_peek_extension.py

```diff
@@ -164,23 +214,40 @@ def test_tablename(schema, expected_tablename):
     assert returned_tablename == expected_tablename
 
 
+@pytest.mark.parametrize(
+    "schema",
+    [
+        ({"x-inherits": 1}),
+        ({"x-open-alchemy-inherits": 1}),
+    ],
+)
 @pytest.mark.helper
-def test_inherits_wrong_type():
+def test_inherits_wrong_type(schema):
     """
     GIVEN schema with x-inherits defined as an integer
     WHEN inherits is called with the schema
     THEN MalformedSchemaError is raised.
     """
-    schema = {"x-inherits": 1}
-
     with pytest.raises(exceptions.MalformedSchemaError):
         helpers.peek.inherits(schema=schema, schemas={})
 
 
 @pytest.mark.parametrize(
     "schema, expected_inherits",
-    [({}, None), ({"x-inherits": "Parent1"}, "Parent1"), ({"x-inherits": True}, True)],
-    ids=["missing", "defined string", "defined boolean"],
+    [
+        ({}, None),
+        ({"x-inherits": "Parent1"}, "Parent1"),
+        ({"x-inherits": True}, True),
+        ({"x-open-alchemy-inherits": "Parent1"}, "Parent1"),
+        ({"x-open-alchemy-inherits": True}, True),
+    ],
+    ids=[
+        "missing",
+        "defined string",
+        "defined boolean",
+        "defined string",
```

Comment:
it might be worth differentiating this id with the other id, maybe mention `x-open-alchemy` or something like that.

The other idea that just came to mind is that you could double-parametrise these tests, first with the different prefixes (`x-` and `x-open-alchemy-`) and then with the different values, it would reduce the number of duplicate information that would need to be defined

Reply:
That's sounds like a very good idea! I'll work on that!

Reply:
I could not get it to work when stacking parametrize decorators to create a matix, but instead I decided to build the `pytest.param` dynamically:

```python
@pytest.mark.parametrize(
    "schema",
    [({f"{prefix}autoincrement": "True"}) for prefix in helpers.peek.VALID_PREFIXES],
)
@pytest.mark.helper
def test_autoincrement_wrong_type(schema):
    """
    GIVEN schema with autoincrement defined as a string
    WHEN autoincrement is called with the schema
    THEN MalformedSchemaError is raised.
    """
    with pytest.raises(exceptions.MalformedSchemaError):
        helpers.peek.autoincrement(schema=schema, schemas={})


@pytest.mark.parametrize(
    "schema, expected_autoincrement",
    [
        pytest.param(params[0], params[1], id=f"{prefix}{params[2]}")
        for prefix in helpers.peek.VALID_PREFIXES
        for params in [
            ({}, None, "missing"),
            ({f"{prefix}autoincrement": True}, True, "true"),
            ({f"{prefix}autoincrement": False}, False, "false"),
        ]
    ],
)
@pytest.mark.helper
def test_autoincrement(schema, expected_autoincrement):
    """
    GIVEN schema and expected autoincrement
    WHEN autoincrement is called with the schema
    THEN the expected autoincrement is returned.
    """
    returned_autoincrement = helpers.peek.autoincrement(schema=schema, schemas={})

    assert returned_autoincrement == expected_autoincrement
```

Let me know what you think of this approach before I move forward with the conversion.

Reply:
I think that is a good start, I had a go at this, what do you think of it? It eliminates a lot of the repetitive code in these test cases and is more scalable with adding additional peek functions:

```python
@pytest.mark.parametrize(
    "prefix",
    [
        pytest.param("x-", id="x-"),
        pytest.param("x-open-alchemy-", id="x-open-alchemy-"),
    ],
)
@pytest.mark.parametrize(
    "key, value, func",
    [
        pytest.param("index", "True", helpers.peek.index, id="index"),
        pytest.param("unique", "True", helpers.peek.unique, id="unique"),
    ],
)
@pytest.mark.helper
def test_peek_value_func_wrong_type(prefix, key, value, func):
    """
    GIVEN prefix, key and value for a schema and function to retrieve the value
    WHEN the function is called with the schema
    THEN MalformedSchemaError is raised.
    """
    schema = {f"{prefix}{key}": value}

    with pytest.raises(exceptions.MalformedSchemaError):
        func(schema=schema, schemas={})


@pytest.mark.parametrize(
    "prefix",
    [
        pytest.param("x-", id="x-"),
        pytest.param("x-open-alchemy-", id="x-open-alchemy-"),
    ],
)
@pytest.mark.parametrize(
    "key_values, func, expected_value",
    [
        pytest.param([], helpers.peek.index, None, id="index missing"),
        pytest.param([("index", True)], helpers.peek.index, True, id="index True"),
        pytest.param([("index", False)], helpers.peek.index, False, id="index False"),
        pytest.param([], helpers.peek.unique, None, id="unique missing"),
        pytest.param([("unique", True)], helpers.peek.unique, True, id="unique True"),
        pytest.param(
            [("unique", False)], helpers.peek.unique, False, id="unique False"
        ),
    ],
)
@pytest.mark.helper
def test_peek_value_func(prefix, key_values, func, expected_value):
    """
    GIVEN prefix and keys and values to build the schema and function to retrieve a
        value
    WHEN the function called with the schema
    THEN the expected value is returned.
    """
    schema = {f"{prefix}{key}": value for key, value in key_values}

    returned_value = func(schema=schema, schemas={})

    assert returned_value == expected_value
```

Reply:
I like that! It should work for most of the tests as long as the schema is not too complicated 👍 

I think the prefix param can still be built dynamically, this way, it will be future proof if we decide to add more prefixes some day :)

```python
@pytest.mark.parametrize(
    "prefix",
    [pytest.param(prefix, id=prefix) for prefix in helpers.peek.VALID_PREFIXES],
)
```